### PR TITLE
add table for Pd FacilitatorProgramRegistrations

### PR DESCRIPTION
**Pull request by Hamms**
As well as FacilitatorTeacherconAttendance, for storing attendance data

### CODE REVIEW

File: dashboard/db/migrate/20170410181336_add_teachercon_to_pd_facilitator_program_registrations.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    add_column :pd_facilitator_program_registrations, :teachercon, :integer
+
+    # make user_id non-nullable
+    change_column_null :pd_facilitator_program_registrations, :user_id, false
+
+    # index on user_id and new teachercon column. Give it a custom name, as the
+    # default-generated one is longer than 64 characters
+    add_index :pd_facilitator_program_registrations, \
+      [:user_id, :teachercon], \
+      unique: true, \
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170410181336_add_teachercon_to_pd_facilitator_program_registrations.rb]: Why are the "\\" characters necessary?

File: dashboard/db/migrate/20170330214255_create_pd_program_registration.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreatePdProgramRegistration < ActiveRecord::Migration[5.0]
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170330214255_create_pd_program_registration.rb]: nit: Ideally, the class name would match the table name.

File: dashboard/db/migrate/20170410181336_add_teachercon_to_pd_facilitator_program_registrations.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/db/migrate/20170410181336_add_teachercon_to_pd_facilitator_program_registrations.rb]: Huh. Apparently they aren't; I could have sworn ruby needed them to escape the line continuations

Comment (ashercodeorg) [reply dashboard/db/migrate/20170410181336_add_teachercon_to_pd_facilitator_program_registrations.rb]: Some method of line continuation is necessary for continuing strings. That what you are thinking of?

**Comment by Hamms** [comment]
No; the reasoning being that the migration can exist without the models, but not vice versa.

**Pull request merged**